src/EFCore.MySql/Infrastructure/Internal/MySqlOptionsExtension.cs

```diff
@@ -55,7 +117,11 @@ public override long GetServiceProviderHashCode()
         {
             if (_serviceProviderHash == null)
             {
-                _serviceProviderHash = (base.GetServiceProviderHashCode() * 397) ^ (ServerVersion?.GetHashCode() ?? 0L);
+                _serviceProviderHash = (base.GetServiceProviderHashCode() * 397) ^
+                                       (ServerVersion?.GetHashCode() ?? 0L) ^
+                                       (NullableCharSetAppendBehavior?.GetHashCode() ?? 0L) ^
+                                       (AnsiCharSetInfo?.GetHashCode() ?? 0L) ^
+                                       (UnicodeCharSetInfo?.GetHashCode() ?? 0L);
```

Comment:
I'm pretty sure that none of these options need to be considered here as they don't affect which services go into the service provider, so you can just return `base.GetServiceProviderHashCode()`

Reply:
@AndriySvyryd these are necessary.  `MySqlOptions` is a singleton and these values get stored in `MySqlOptions`.  The tests failed with race conditions when I removed this.